File: clickyman-common/src/main/java/com/clickyman/entities/BaseEntity.java
package com.clickyman.entities;

import java.util.Date;
import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;
import javax.persistence.PrePersist;

import org.hibernate.annotations.GenericGenerator;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@MappedSuperclass
public class BaseEntity {
	@Id
	@GeneratedValue(generator = "uuid2")
	@GenericGenerator(name = "uuid2", strategy = "uuid2")
	private UUID id;
	
	@Column(nullable = false)
	private Date createdDate;
	
	@Column(nullable = false)
	private Date updatedDate;
	
	@PrePersist
	public void beforeInsert() {
		this.createdDate = new Date();
		this.updatedDate = new Date();
	}
	
	public void beforeUpdate() {
		this.updatedDate = new Date();
	}
}

File: post-service/src/entities/base.entity.ts
import {AfterUpdate, BeforeInsert, Column, PrimaryGeneratedColumn} from "typeorm";
import {Type} from "class-transformer";
import {IsDate} from "class-validator";

export abstract class BaseEntity {
	@PrimaryGeneratedColumn("uuid")
	public id!: string;

	@Column()
	@Type(() => Date)
	@IsDate()
	public createdDate: Date;

	@Column()
	@Type(() => Date)
	@IsDate()
	public updatedDate: Date;

	@BeforeInsert()
	protected generateDateBeforeInsert(): void {
		this.createdDate = new Date();
		this.updatedDate = new Date();
	}

	@AfterUpdate()
	protected generateDateAfterUpdate(): void {
		this.updatedDate = new Date();
	}
}
